### PR TITLE
Add textScaleFactor to SelectableText

### LINKED ISSUE #44674: Missing textScaleFactor argument in SelectableText

### PR DESCRIPTION
**Pull request by KagurazakaHanabi**
## Related Issues

Closes #44674 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by darielopper — APPROVED** [review]

File: packages/flutter/lib/src/material/selectable_text.dart [hunk 1]
```diff
@@ -48,7 +48,7 @@ class _SelectableTextSelectionGestureDetectorBuilder extends TextSelectionGestur
   _SelectableTextSelectionGestureDetectorBuilder({
     @required _SelectableTextState state,
   }) : _state = state,
-      super(delegate: state);
+        super(delegate: state);
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/selectable_text.dart]: nit: remove one space from the indentation.

File: packages/flutter/lib/src/material/selectable_text.dart [hunk 2]
```diff
@@ -391,6 +396,7 @@ class SelectableText extends StatefulWidget {
     properties.add(IntProperty('maxLines', maxLines, defaultValue: null));
     properties.add(EnumProperty<TextAlign>('textAlign', textAlign, defaultValue: null));
     properties.add(EnumProperty<TextDirection>('textDirection', textDirection, defaultValue: null));
+    properties.add(DoubleProperty('textScaleFactor', textScaleFactor, defaultValue: 1.0));
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/selectable_text.dart]: From the constructor: The default value seems to be null.

**Review by goderbauer — COMMENTED** [review]
The change looks ok. However, this will need a test before we can merge it.

Also: Please merge in the latest master to make the Cirrus CI happy.

/cc @chunhtai

**Comment by KagurazakaHanabi** [comment]
@goderbauer All done

**Review by goderbauer — COMMENTED** [review]
This will need a test that shows that the textScaleFactor is actually respected. The current test only tests that it's printed out as part of the diagnostics.

**Comment by KagurazakaHanabi** [comment]
Sorry i don't know how to write it...

**Comment by chunhtai** [comment]
@KagurazakaHanabi 

I think this is something you can try
```dart
testWidgets('has expected defaults', (WidgetTester tester) async {
    await tester.pumpWidget(
      const MediaQuery(
        data: MediaQueryData(devicePixelRatio: 1.0),
        child: Directionality(
          textDirection: TextDirection.ltr,
          child: SelectableText('selectable text', textSacleFactor: 1.3),
        ),
      ),
    );

   RenderBox box = tester.renderObject(find.byType(SelectableText));
    expect(box.size.height, <some expected value>);
    await tester.pumpWidget(
      const MediaQuery(
        data: MediaQueryData(devicePixelRatio: 1.0),
        child: Directionality(
          textDirection: TextDirection.ltr,
          child: SelectableText('selectable text', textSacleFactor: 1.9),
        ),
      ),
    );
  });
  box = tester.renderObject(find.byType(SelectableText));
    expect(box.size.height, <some different expected value>);
```

File: packages/flutter/test/widgets/selectable_text_test.dart [hunk 3]
```diff
@@ -192,22 +189,32 @@ void main() {
...
-    final SelectableText selectableText =
-    tester.firstWidget(find.byType(SelectableText));
+    final SelectableText selectableText = tester.firstWidget(find.byType(SelectableText));
     expect(selectableText.showCursor, false);
     expect(selectableText.autofocus, false);
     expect(selectableText.dragStartBehavior, DragStartBehavior.start);
     expect(selectableText.cursorWidth, 2.0);
     expect(selectableText.enableInteractiveSelection, true);
+
+    RenderBox renderBox = tester.renderObject(find.byType(SelectableText));
```
Comment (chunhtai) [inline packages/flutter/test/widgets/selectable_text_test.dart]: Let's separate this part into a different test. The original test is to test it has expected default values, while this part is testing the box size changes based on textScaleFactor.

Comment (KagurazakaHanabi) [reply packages/flutter/test/widgets/selectable_text_test.dart]: done

File: packages/flutter/test/widgets/selectable_text_test.dart [hunk 4]
```diff
@@ -3194,6 +3209,7 @@ void main() {
 //    properties.add(IntProperty('maxLines', maxLines, defaultValue: null));
 //    properties.add(EnumProperty<TextAlign>('textAlign', textAlign, defaultValue: null));
 //    properties.add(EnumProperty<TextDirection>('textDirection', textDirection, defaultValue: null));
+//    properties.add(DoubleProperty('textScaleFactor', textScaleFactor, defaultValue: null));
```
Comment (chunhtai) [inline packages/flutter/test/widgets/selectable_text_test.dart]: I think i accidentally include this when I wrote the test, can you remove all the comments?

**Review by chunhtai — APPROVED** [review]
LGTM with one minor comment. Sorry for the long wait.

File: packages/flutter/test/widgets/selectable_text_test.dart (hunk 4 above)
Comment (KagurazakaHanabi) [reply packages/flutter/test/widgets/selectable_text_test.dart]: Done

**Pull request merged**